src/Symfony/Component/Mailer/Bridge/Postal/Transport/PostalApiTransport.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Postal\Transport;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\Exception\HttpTransportException;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\AbstractApiTransport;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+final class PostalApiTransport extends AbstractApiTransport
+{
+    public function __construct(
+        #[\SensitiveParameter] private string $apiToken,
+        ?HttpClientInterface $client = null,
+        ?EventDispatcherInterface $dispatcher = null,
+        ?LoggerInterface $logger = null,
+    ) {
+        parent::__construct($client, $dispatcher, $logger);
+    }
+
+    public function __toString(): string
+    {
+        return \sprintf('postal+api://%s', $this->getEndpoint());
+    }
+
+    protected function doSendApi(SentMessage $sentMessage, Email $email, Envelope $envelope): ResponseInterface
+    {
+        $path = \sprintf('/api/v1/send/message');
+
+        $response = $this->client->request('POST', 'https://'.$this->getEndpoint().$path, [
```

Comment:
How does this work when the host is `null`?

Reply:
It probably won't. As Postal is a self-hosted service, the expectation is that users always set the host. Should I implement a check to cover this, and throw an exception if it isn't set?

Reply:
The `PostalTransportFactory` should account for this then. Currently it contains the following code:

```php
$host = 'default' === $dsn->getHost() ? null : $dsn->getHost();`
```

Reply:
Gotcha. I'll implement it! :)